### ISSUE TITLE
[Linux, Screen] Radiant Menu Location Follows Mouse Even When Closed

### ISSUE THREAD

**Issue by yoshiyoshyosh** [Yellow-Dog-Man/Resonite-Issues#47]
### Describe the bug?

On screen mode in native Linux, the position of the radiant menu appears to follow the movement of the mouse when it is closed. This behavior is not observed on Proton.

### To Reproduce

1. Launch the client on Linux Native screen mode
2. Move mouse
3. Open radiant menu

### Expected behavior

The radiant menu appears in the middle of the screen, like Proton.

### Screenshots

https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/36683489/a842e9c0-3271-4a52-9e51-6d83ed5cec35

### Resonite Version Number

2023.10.13.743

### What Platforms does this occur on?

Linux

### Log Files

[Resonite log](https://gist.github.com/yoshiyoshyosh/aa26de760e3a1c53427d7a7dd3d4c06d)

[Unity Player.log](https://gist.github.com/yoshiyoshyosh/ce9f75ce1603eab607b0e18232b14cad)

### Reporters

yosh

**Comment by MegaShinySnivy** [comment]
Can replicate on vanilla SteamOS in a VM, along with an arch linux native system on wayland

**Comment by Frooxius** [comment]
This is a bit of a tricky issue. Unity on Linux doesn't seem to support locking the cursor, so our request to lock the cursor are simply ignored. 

I don't think we'll be able to address this easily until we switch away from Unity.

**Comment by ljoonal** [comment]
> Unity on Linux doesn't seem to support locking the cursor

I'm not sure what exactly isn't supposedly supported... Starting with Unity 2019.2 [the documentation](https://docs.unity3d.com/2019.2/Documentation/ScriptReference/Cursor-lockState.html) explicitly states that it _is_ supported on linux:

> This is only supported on Windows and Linux standalone builds.

And even if _somehow_ it isn't, [unlocking the cursor and then calling `Mouse.current.WarpCursorPosition`](https://github.com/ljoonal/Resonite-Mods/blob/8838c6f3c70659ac29cbefe9d6255d96a30f2939/LinuxFixes/LinuxFixes.cs#L53) has worked for me for ages, dating back to NeosVR... (and in case lawyers think such a little code snippet is somehow copyrightable, I'll grant Frooxius/YellowManStudios the linked file's code under a [CC0](https://creativecommons.org/public-domain/cc0/)/[MIT](https://mit-license.org/) license :P ). Though of course `worksOnMyMachineTM` doesn't mean it'll work across all distros I guess...

**Comment by Frooxius** [comment]
The reason why we state that is because the Linux build is using exact same codepath and API as on Windows, but it just doesn't work. We are not excluding or skipping the API while on Linux - same things are being called as on Windows, but the issue is still present.

The lockCursor position does seem to work on Linux, but the other part - WarpCursorPosition, which is responsible for re-centering the cursor does not seem to do anything.

The code snippet is a bit odd, since it seems to run this in the context menu? Rather than the mouse driver, which normally handles this generically for Resonite.

Are you able to actually get the WarpCursorPosition to position the cursor arbitrarily on Linux?

**Comment by ljoonal** [comment]
> Are you able to actually get the WarpCursorPosition to position the cursor arbitrarily on Linux?

Yes. I posted videos to the official resonite #linux-build channel, [a video without the `LinuxFixes` mod](https://cdn.discordapp.com/attachments/1183927772700213299/1200596072142229554/before.mp4?ex=65c6c139&is=65b44c39&hm=403b810497c003b7f18760671fd09c14cb44f1d95a8f3ae82ad91177bc392616&), [and a video with it](https://cdn.discordapp.com/attachments/1183927772700213299/1200596072804917289/after.mp4?ex=65c6c139&is=65b44c39&hm=471417227496c7833a476daefb0bcfb07bda696f4f520d487fd2a533722aa5ec&).

So it does mostly fix it on my system at least (used to be i3 during Neos, now hyprland), although towards the end can see that when am already moving my mouse whilst menu is still closing the menu opens to a weird place still, so it does still have a few issues when being too fast. But works in 99% of the cases (aka not trying to spam-open the menu).

**Comment by shiftyscales** [comment]
Is this issue still relevant even when running the Windows build through Proton,@yoshiyoshyosh? If not, this can probably be closed for now as the native Linux client has been deprecated indefinitely.

**Comment by yoshiyoshyosh** [comment]
> Is this issue still relevant even when running the Windows build through Proton,[@yoshiyoshyosh](https://github.com/yoshiyoshyosh)? If not, this can probably be closed for now as the native Linux client has been deprecated indefinitely.

no. it is also a unity issue, so it'll most likely go away once we're off unity

**Comment by shiftyscales** [comment]
Thank you, closing.
